plugins/dominant-color-images/hooks.php

```diff
@@ -186,3 +186,80 @@ function dominant_color_render_generator(): void {
 	echo '<meta name="generator" content="dominant-color-images ' . esc_attr( DOMINANT_COLOR_IMAGES_VERSION ) . '">' . "\n";
 }
 add_action( 'wp_head', 'dominant_color_render_generator' );
+
+/**
+ * Adds inline CSS for dominant color styling in the WordPress admin area.
+ *
+ * This function registers and enqueues a custom style handle, then adds inline CSS
+ * to apply background color based on the dominant color for attachment previews
+ * in the WordPress admin interface.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_inline_style(): void {
+	$handle = 'dominant-color-admin-styles';
+	// PHPCS ignore reason: Version not used since this handle is only registered for adding an inline style.
+	// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = '.wp-core-ui .attachment-preview[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_action( 'admin_enqueue_scripts', 'dominant_color_admin_inline_style' );
+
+/**
+ * Adds a script to the admin footer to modify the attachment template.
+ *
+ * This function injects a JavaScript snippet into the admin footer that modifies
+ * the attachment template. It adds attributes for dominant color and transparency
+ * to the template, allowing these properties to be displayed in the media library.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_script(): void {
+	?>
+	<script type="module">
+		const tmpl = document.getElementById( 'tmpl-attachment' );
+		if ( tmpl ) {
+			tmpl.textContent = tmpl.textContent.replace(
+				'{{ data.orientation }}"',
+				'{{ data.orientation }} {{ data.hasTransparency ? \'has-transparency\' : \'not-transparent\' }}"	data-dominant-color="{{ data.dominantColor }}" data-has-transparency="{{ data.hasTransparency }}" style="--dominant-color: #{{ data.dominantColor }};"'
+			);
```

Comment:
As I understand, this will be injecting additional attributes at the end of this `DIV`: https://github.com/WordPress/wordpress-develop/blob/0d428194d1e6d6f1285d45cafd84b18f514070c1/src/wp-includes/media-template.php#L596

This seems like it could be a bit brittle, like what if someone has that text string show up somewhere else? Or what if another plugin already appended the `class` with some other text than `{{ data.orientation }}`? Would it be better to instead just look for the first `DIV` and then append the attributes? So something like this:

```suggestion
		tmpl.textContent = tmpl.textContent.replace( /^\s*<div[^>]*?(?=>)/, ( match ) => {
			let replaced = match.replace( /\sclass="/, " class=\"{{ data.hasTransparency ? \'has-transparency\' : \'not-transparent\' }} " );
			replaced += ' data-dominant-color="{{ data.dominantColor }}"';
			replaced += ' data-has-transparency="{{ data.hasTransparency }}"';
			replaced += ' style="--dominant-color: #{{ data.dominantColor }};"'; // TODO: Potentially there could be a style attribute as well!
			return replaced;
		} );
```

Granted, this is probably not going to happen.

Too bad we don't have the HTML Tag Processor in JavaScript!